python/tests/kernel/test_kernel_float_complex.py

```diff
@@ -0,0 +1,118 @@
+# ============================================================================ #
+# Copyright (c) 2022 - 2024 NVIDIA Corporation & Affiliates.                   #
+# All rights reserved.                                                         #
+#                                                                              #
+# This source code and the accompanying materials are made available under     #
+# the terms of the Apache License 2.0 which accompanies this distribution.     #
+# ============================================================================ #
+
+import os, sys
+
+import pytest
+import numpy as np
+import math
+from typing import List 
+
+import cudaq
+from cudaq import spin
+
+## [PYTHON_VERSION_FIX]
+skipIfPythonLessThan39 = pytest.mark.skipif(
+    sys.version_info < (3, 9),
+    reason="built-in collection types such as `list` not supported")
+
+def test_float():
+    """Test that we can use float numbers inside kernel functions."""
+
+    cudaq.reset_target()
+    cudaq.set_target('nvidia-fp64')
+
+    f = [0., 1., 1., 0.]
+
+    # Pass a list of float as a parameter
+    @cudaq.kernel
+    def test_float_vec_param(vec : list[float]):
+        f1 = vec
+
+    counts = cudaq.sample(test_float_vec_param, f)
+    assert len(counts) == 0
+
+
+    # Capture a list of float
+    @cudaq.kernel
+    def test_float_vec_capture():
+        f1 = f
+
+    counts = cudaq.sample(test_float_vec_capture)
+    assert len(counts) == 0
+
+
+    # Define a list of float inside a kernel
+    @cudaq.kernel
+    def test_float_vec_definition():
+        f1 = [1.0, 0., 0., 1.]
+
+    counts = cudaq.sample(test_float_vec_definition)
+    assert len(counts) == 0
+
+    # Use a float inside a kernel
+    @cudaq.kernel
+    def test_float_use():
+        f1 = math.sin(0)
+
+    counts = cudaq.sample(test_float_use)
+    assert len(counts) == 0
+
+    # Use a float inside np in a kernel
+    @cudaq.kernel
+    def test_float_np_use():
+        f1 = np.sin(0)
+
+    counts = cudaq.sample(test_float_np_use)
+    assert len(counts) == 0
+
+def test_complex():
```

Comment:
I would suggest also adding support for `c = np.array([...], dtype=np.complex64)` and demonstrate with the `nvidia` target. Also it would be good to test that we raise an exception for the use of `complex` with `nvidia` and `np.complex64` with FP64 backends (all the other ones).

Reply:
I created a separate issue for supporting np arrays: https://github.com/NVIDIA/cuda-quantum/issues/1523, will work in it in the next PR